xarray/backends/common.py

```diff
@@ -369,10 +369,32 @@ class BackendEntrypoint:
     - ``guess_can_open`` method: it shall return ``True`` if the backend is able to open
       ``filename_or_obj``, ``False`` otherwise. The implementation of this
       method is not mandatory.
+
+    Attributes
+    ----------
+
+    open_dataset_parameters : tuple, default None
+        A list of ``open_dataset`` method parameters.
+        The setting of this attribute is not mandatory.
+    description : str
+        A short string describing the engine.
+        The setting of this attribute is not mandatory.
+    url : str
+        A string with the URL to the backend's documentation.
+        The setting of this attribute is not mandatory.
     """
 
     open_dataset_parameters: tuple | None = None
-    """list of ``open_dataset`` method parameters"""
+    description: str = ""
+    url: str = ""
+
+    def __str__(self) -> str:
+        txt = f"Backend type: {type(self).__name__}"
+        if self.description:
+            txt += f"\n{self.description}"
+        if self.url:
+            txt += f"\nLearn more at {self.url}"
+        return txt
```

Comment:
```suggestion

    def __repr__(self) -> str:
        description = "Base entrypoint for backends"
        url = r"https://xarray.dev/"
        return f"<{type(self).__name__}>\n  {description}\n  {url}"
```
Here's how I would format it. I don't think the attributes are necessary and can live in the repr. The entrypoints xarray support should be updated with meaningful descriptions and links.

```python
# The single repr:
xr.backends.list_engines()["zarr"]
Out[19]: 
<ZarrBackendEntrypoint>
  Base entrypoint for backends
  https://xarray.dev/

# list_engines is quite readable with the indents:
xr.backends.list_engines()
Out[18]: 
{'netcdf4': <NetCDF4BackendEntrypoint>
   Base entrypoint for backends
   https://xarray.dev/,
 'h5netcdf': <H5netcdfBackendEntrypoint>
   Base entrypoint for backends
   https://xarray.dev/,
 'scipy': <ScipyBackendEntrypoint>
   Base entrypoint for backends
   https://xarray.dev/,
 'pseudonetcdf': <PseudoNetCDFBackendEntrypoint>
   Base entrypoint for backends
   https://xarray.dev/,
 'pydap': <PydapBackendEntrypoint>
   Base entrypoint for backends
   https://xarray.dev/,
 'store': <StoreBackendEntrypoint>
   Base entrypoint for backends
   https://xarray.dev/,
 'zarr': <ZarrBackendEntrypoint>
   Base entrypoint for backends
   https://xarray.dev/}
```

Reply:
But then all external backend will have to re-implement the __repr__ and might not follow the indentation. Personally I like the attributes approach more.

Reply:
I think that's okay, it's easy enough to copy/paste the base repr. 
I suspect the `url` will be a wasted attribute as some engines might not have a homepage.

Reply:
I took an approach somewhere in between. If we make the descriptions part of the base repr instead of attributes, then it will either have filler text (as above) or funny spacing (if we let `description/url = ""`) if each Backend doesn't fill in the repr fields. I did change it to `__repr__` and add the nice indentation (though the dict doesn't print with newlines after each key in my output window, which makes it slightly confusing to look at):
```
{'netcdf4': <NetCDF4BackendEntrypoint>, 'scipy': <ScipyBackendEntrypoint>
  this is how you use scipy as a backend
  Learn more at scipy.org, 'pydap': <PydapBackendEntrypoint>, 'rasterio': <RasterioBackend>, 'store': <StoreBackendEntrypoint>, 'zarr': <ZarrBackendEntrypoint>}
```

Happy to make further adjustments but trying to include all suggestions!

Reply:
I think whether or not the description and url live in the attributes is not a big deal, and it would be great to just get get this merged as is so we can include it in the upcoming release! :grinning:  

> (though the dict doesn't print with newlines after each key in my output window, which makes it slightly confusing to look at)

I think you're going to have this problem with any solution that still uses a dict of reprs. (Not sure why that doesn't seem to be the case in @Illviljan's example though.)